src/Swift.Runtime/src/Metadata/TypeMetadata.cs

```diff
@@ -249,4 +253,113 @@ public override int GetHashCode ()
+
+        // TODO: handle existential containers https://github.com/dotnet/runtimelab/issues/2875
+
+        result = null;
+        return false;
+    }
+
+    /// <summary>
+    /// Returns an enumeration of known Type and TypeMetadata objects
+    /// </summary>
+    /// <returns>An enumeration of known Type and TypeMetadata objects</returns>
+    /// <exception cref="Exception">Throws if unable to load a library containing known types</exception>
+    static IEnumerable<(Type, TypeMetadata)> KnownMetadata()
+    {
+        var libraryHandle = NativeLibrary.Load(KnownLibraries.SwiftCore);
+        if (libraryHandle == IntPtr.Zero)
+            throw new Exception(string.Format("Unable to load library {0}", KnownLibraries.SwiftCore));
+        // types from libSwiftCore
+        yield return (typeof(bool), MetadataFromNativeLibrary(libraryHandle, "$sSbN"));
+        yield return (typeof(nint), MetadataFromNativeLibrary(libraryHandle, "$sSiN"));
+        yield return (typeof(nuint), MetadataFromNativeLibrary(libraryHandle, "$sSuN"));
+        yield return (typeof(float), MetadataFromNativeLibrary(libraryHandle, "$sSfN"));
+        yield return (typeof(double), MetadataFromNativeLibrary(libraryHandle, "$sSdN"));
+        yield return (typeof(sbyte), MetadataFromNativeLibrary(libraryHandle, "$ss4Int8VN"));
+        yield return (typeof(byte), MetadataFromNativeLibrary(libraryHandle, "$ss5UInt8VN"));
+        yield return (typeof(short), MetadataFromNativeLibrary(libraryHandle, "$ss5Int16VN"));
+        yield return (typeof(ushort), MetadataFromNativeLibrary(libraryHandle, "$ss6UInt16VN"));
+        yield return (typeof(int), MetadataFromNativeLibrary(libraryHandle, "$ss5Int32VN"));
+        yield return (typeof(uint), MetadataFromNativeLibrary(libraryHandle, "$ss6UInt32VN"));
+        yield return (typeof(long), MetadataFromNativeLibrary(libraryHandle, "$ss5Int64VN"));
+        yield return (typeof(ulong), MetadataFromNativeLibrary(libraryHandle, "$ss6UInt64VN"));
+        yield return (typeof(void), MetadataFromNativeLibrary(libraryHandle, "$sytN"));
+
+        // to add more primitives from different libraries, reassign libraryHandle and yield return
+        // the symbols starting from here.
+    }
+
+    /// <summary>
+    /// Loads type metadata from a NativeLibrary using the supplied symbol
+    /// </summary>
+    /// <param name="handle">handle to a library loaded by NativeLibrary</param>
+    /// <param name="symbolName">Swift symbol for a type metadata object</param>
+    /// <param name="libraryName">The library to load from. Defaults to libswiftCore.dylib</param>
+    /// <returns>A type metadata object for the symbol</returns>
+    /// <exception cref="Exception">Throws on failure to load symbol</exception>
+    static TypeMetadata MetadataFromNativeLibrary(IntPtr handle, string symbolName, string libraryName = KnownLibraries.SwiftCore)
+    {
+        var entryPoint = NativeLibrary.GetExport(handle, symbolName);
```

Comment:
[`NativeLibrary.GetExport()`](https://learn.microsoft.com/dotnet/api/system.runtime.interopservices.nativelibrary.getexport#system-runtime-interopservices-nativelibrary-getexport(system-intptr-system-string)) throws if the export can't be found. Instead `TryGetExport()` should be used if a specific exception is desired.